### PR TITLE
Added/Updated paramater objects for S3. Also added existing S3 methods

### PR DESCRIPTION
**Pull request by nickzelei**
case 1. Add a new type definition.
- [ ] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [ ] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [ ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

File directory modified: [AWS-SDK](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/aws-sdk)

### CODE REVIEW

File: aws-sdk/aws-sdk.d.ts [hunk 1]
```diff
@@ -156,6 +156,14 @@ declare module "aws-sdk" {
 		constructor(options?: any);
 		putObject(params: s3.PutObjectRequest, callback: (err: any, data: any) => void): void;
 		getObject(params: s3.GetObjectRequest, callback: (err: any, data: any) => void): void;
+    putObject(params: s3.PutObjectRequest, callback: (err: Error, data: any) => void): void;
```
Comment (RyanCavanaugh) [inline aws-sdk/aws-sdk.d.ts]: Please format these with spacing consistent with the surrounding block

File: aws-sdk/aws-sdk.d.ts [hunk 2]
```diff
@@ -156,6 +156,14 @@ declare module "aws-sdk" {
...
+    getObject(params: s3.GetObjectRequest, callback: (err: Error, data: any) => void): void;
+    deleteObject(params: s3.DeleteObjectRequest, callback: (err: Error, data: any) => void): void;
+    headObject(params: s3.HeadObjectRequest, callback: (err: Error, data: any) => void): void;
+    getSignedUrl(operation: string, params: any): string;
+    getSignedUrl(operation: string, params?: any, callback?: (err: Error, url: string) => void): void;
+    upload(params: s3.PutObjectRequest, options?: s3.UploadOptions): void;
```
Comment (RyanCavanaugh) [inline aws-sdk/aws-sdk.d.ts]: This line can be removed as the signature below is strictly more useful

File: aws-sdk/aws-sdk.d.ts [hunk 3]
```diff
@@ -156,6 +156,14 @@ declare module "aws-sdk" {
 		constructor(options?: any);
 		putObject(params: s3.PutObjectRequest, callback: (err: any, data: any) => void): void;
 		getObject(params: s3.GetObjectRequest, callback: (err: any, data: any) => void): void;
+    putObject(params: s3.PutObjectRequest, callback: (err: Error, data: any) => void): void;
+    getObject(params: s3.GetObjectRequest, callback: (err: Error, data: any) => void): void;
+    deleteObject(params: s3.DeleteObjectRequest, callback: (err: Error, data: any) => void): void;
+    headObject(params: s3.HeadObjectRequest, callback: (err: Error, data: any) => void): void;
+    getSignedUrl(operation: string, params: any): string;
+    getSignedUrl(operation: string, params?: any, callback?: (err: Error, url: string) => void): void;
```
Comment (RyanCavanaugh) [inline aws-sdk/aws-sdk.d.ts]: According to the docs these parameters should not be optional

File: aws-sdk/aws-sdk.d.ts [hunk 4]
```diff
@@ -146,16 +146,23 @@ declare module "aws-sdk" {
...
 	}
 
 	export class S3 {
-		constructor(options?: any);
-		putObject(params: s3.PutObjectRequest, callback: (err: any, data: any) => void): void;
-		getObject(params: s3.GetObjectRequest, callback: (err: any, data: any) => void): void;
+        constructor(options?: any);
+        putObject(params: s3.PutObjectRequest, callback: (err: any, data: any) => void): void;
+        getObject(params: s3.GetObjectRequest, callback: (err: any, data: any) => void): void;
+        putObject(params: s3.PutObjectRequest, callback: (err: Error, data: any) => void): void;
```
Comment (RyanCavanaugh) [inline aws-sdk/aws-sdk.d.ts]: This overload is unobservable because the one two lines above is always chosen instead.

File: aws-sdk/aws-sdk.d.ts [hunk 5]
```diff
@@ -146,16 +146,23 @@ declare module "aws-sdk" {
...
+        getObject(params: s3.GetObjectRequest, callback: (err: Error, data: any) => void): void;
```
Comment (RyanCavanaugh) [inline aws-sdk/aws-sdk.d.ts]: Same here

**Comment by zelein** [comment]
There hasn't been any activity on this request in the last week. Is there any way I can get an update on this so I can know if there is anything else that I need to do?

**Pull request merged**